Medical Nerf: Handheld Adventures

Geevies opened the pull request:
Modifies the Handheld Health Analyzer's readouts a little. 

Firstly, it displays if someone is malnourished or dehydrated.
Secondly, it changes the numerical damage values you get if you scan someone into words, this prevents doctors from screaming "OH YOU HAVE 0.3 TOXINS", which doesn't really make sense in any term of lore standards. I guess. Lmao.
Thirdly, it removes the weird "Anatomical damage detected" lines, because those are literally just repeats of the damage values you're already getting.

VTCobaltblood commented:
> Secondly, it changes the numerical damage values you get if you scan someone into words, this prevents doctors from screaming "OH YOU HAVE 0.3 TOXINS", which doesn't really make sense in any term of lore standards. I guess. Lmao.

Wouldn't that weirdly make sensors more elaborate than scanners? Also, isn't it a bit weird to describe someone's damage as critical if the person is already dead? Like, maybe make values above 200 be described as "fatal"? 

~~New descriptors every hundred damage for batshit people?~~

Geevies commented:
Gonna add fatal, gimme a second.

Geevies commented:
It would make suit sensors more accurate than Handheld Scanners yes, but Suit Sensors' stuff's done in HTML, which I'm woefully bad at. I might wrangle Karolis or PoZe or Skull or someonepleasedoitidon'twannalearnHTML

It's on the to-do list.

Geevies commented:
Okay, it was brought to my attention that scientists won't be able to test how good their guns are on monkeys are anymore, so I allowed monkeys (only monkeys, not there other lesser forms) to be placed into the Adv. Body Scanners.

Happy hunting.

OneOneThreeEight commented:
REEE UNATOMIC PR.. reeee!

No, but, I suppose these changes are sensible. They shouldn't be as accurate as stationary scanners.

Geevies commented:
I can separate it into two PRs if needs be, but uh. I'm not sure why monkeys weren't allowed into scanners in the first place.

Also, die.

Geevies commented:
!review

Arrow768 commented:
Could you post a screenshot how the scanner output looks now and how it looks with fake death (i.e. Changeling Regenerative Stasis)

Geevies commented:
Can do.

Healthy:
![image](https://user-images.githubusercontent.com/22774890/59153262-af55cd80-8a55-11e9-9995-4cee99dca1a6.png)

Major damage + Bleeding; Chest:
![image](https://user-images.githubusercontent.com/22774890/59153267-c694bb00-8a55-11e9-8d04-1eb362c2e8f7.png)

Healthy + Malnourished + Dehydrated:
![image](https://user-images.githubusercontent.com/22774890/59153276-e3c98980-8a55-11e9-83b2-ecf30346a724.png)

Toxin + Brute + Localized Damage + Malnourished + Dehydrated:
![image](https://user-images.githubusercontent.com/22774890/59153306-2db26f80-8a56-11e9-97d6-c5d32e292056.png)

Geevies commented:
Changeling too.

Faked death:
![image](https://user-images.githubusercontent.com/22774890/59153314-681c0c80-8a56-11e9-80d5-be2fb800a081.png)

Faked death + Injured Chest + Dehydrated + Malnourished:
![image](https://user-images.githubusercontent.com/22774890/59153318-926dca00-8a56-11e9-8eac-fd1ff5e1a9b0.png)

Geevies commented:
Latest commit makes the actual code look like the screenshots produced. Removed the strings after the spaces and added it before the \[Bleeding\] text instead, so that they wouldn't be squished together.

Geevies commented:
Alright, I modified medical PDA cartridges to follow the same ruleset. And I added an advanced medical scanner to RnD that functions the same as the old one, but you need to do some research to get it.

I was also held at gunpoint to use a Citadel RP sprite, I take no responsibility, also your sprite is sleek as fuck. Thanks.

VTCobaltblood commented on code/modules/research/designs/medical_designs.dm:
```diff
@@ -10,12 +10,22 @@
 	name = "Biotech device prototype ([item_name])"
 
 /datum/design/item/health_analyzer
+	name = "Health Analyzer"
```
This should be a subclass of medical designs, with a lowercase name.

Geevies replied:
Donezo. It's still camelcase because that fits with the other medical items in the research designs doc.

VTCobaltblood commented on code/modules/research/designs/medical_designs.dm:
```diff
@@ -10,12 +10,22 @@
 	build_path = /obj/item/device/healthanalyzer
 	sort_string = "MBCAF"
+	
+/datum/design/item/adv_health_analyzer/
```
This too.

Geevies replied:
Ditto.

Geevies commented:
Suggestions added. Seems the person who previously put analyzers in forgot to put them under medical. Technically a bugfix PR now too!

Arrow768 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -23,26 +23,43 @@ BREATH ANALYZER
-		user.show_message("<span class='notice'>Key: Suffocation/Toxin/Burns/Brute</span>", 1)
-		user.show_message("<span class='notice'>Body Temperature: ???</span>", 1)
-		return
+			user.show_message("<span class='notice'>Analyzing Results for The floor:</span>", 1)
```
If you add that indentation there, then the user will get the Analyzing results for the floor one time for every mob that gets returned by viewers.
Not sure if thats the goal here.

Geevies replied:
Good catch, fixing now.

Arrow768 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -23,26 +23,43 @@ BREATH ANALYZER
+/proc/health_scan_mob(var/mob/living/M, var/mob/living/user, var/adv, var/visible_msg, var/ignore_clumsiness, var/show_limb_damage = TRUE)
+	if (((user.is_clumsy()) || (DUMB in user.mutations)) && prob(50))
 		to_chat(user, text("<span class='warning'>You try to analyze the floor's vitals!</span>"))
 		for(var/mob/O in viewers(M, null))
```
Get rid of that and the line below and replace it with that proc
`/mob/visible_message(var/message, var/self_message, var/blind_message, var/range = world.view)`

Geevies replied:
Donezo.

Geevies commented:
Had a breakthrough when working on one of my other PRs. The Crew Monitoring console now works under the same variables as the handheld scanner does, albeit the "Healthy" cap a bit raised to allow some leeway. (Two punches will still display Healthy. Three punches or a stab wound will display Minor)

![image](https://user-images.githubusercontent.com/22774890/59198111-3fe5f800-8b93-11e9-946c-5e6f452746c9.png)
